config/initializers/core_extensions.rb

```diff
@@ -0,0 +1,2 @@
+# frozen_string_literal: true
+Dir[File.join(Rails.root, 'lib', 'core_extensions', '*.rb')].each { |file| require file }
```

Comment:
how about putting immutable_string in lib or namespacing it with `module CoreExtensions` so that it gets autoloaded?